src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -290,7 +292,7 @@ private function getAutowiredReference(TypedReference $reference, $deprecationMe
             return new TypedReference($this->types[$type], $type);
```

Comment:
shouldn't we have a deprecation warning triggered here, to tell people to register the type for the new system rather than relying on old types (as you disable the loading in strict mode)

Reply:
They will get an exception saying that. Not sure there is anything else needed.

Reply:
@stof I re-enabled by-explicit-autowiring-type in strict mode, that should address you concern, isn't it?